src/main/java/redis/clients/jedis/JedisCluster.java

```diff
@@ -222,6 +223,10 @@ private JedisCluster(ClusterConnectionProvider provider, int maxAttempts, Durati
     super(provider, maxAttempts, maxTotalRetriesDuration, protocol);
   }
 
+  public JedisCluster(ClusterCommandExecutor executor, ClusterConnectionProvider provider, ClusterCommandObjects commandObjects, RedisProtocol protocol) {
```

Comment:
1. We do not plan to change CommandExecutor from here.
```suggestion
  public JedisCluster(ClusterConnectionProvider provider, ClusterCommandObjects commandObjects, RedisProtocol protocol) {
```
2. Add a JavaDoc stating redisProtocol will be set into commandObjects.